Minio won't start due to "invalid memory address or nil pointer dereference"

Minio won't start in latest version on docker swarm (image minio:latest) since today. The previous release is fine.

We're using AssumeRoleWithWebidentity/the MINIO_IAM_JWKS_URL setting, so that might cause the issue.

Here's the log
[log.txt](https://github.com/minio/minio/files/3716942/log.txt)

@vadmeste After setting a fixed version (RELEASE.2019-10-11T00-38-09Z), the issue is still the same:

```
# docker ps
CONTAINER ID        IMAGE                                       COMMAND                  CREATED             STATUS                           PORTS               NAM                                                                        ES
7df9ab1982e6        minio/minio:RELEASE.2019-10-11T00-38-09Z    "/usr/bin/docker-ent…"   12 seconds ago      Up 1 second (health: starting)   9000/tcp            min                                                                        io_node3.1.8q6otlkylzatc0pbfnql15n4j
55897abd4a31        minio/minio:RELEASE.2019-10-11T00-38-09Z    "/usr/bin/docker-ent…"   2 minutes ago       Up 2 minutes (healthy)           9000/tcp            min                                                                        io_node4.1.z2quzl6q1fl9nizbozer6vdq2

lmanlxapp42:/root
# docker logs 7d
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x45ee88]

goroutine 1 [running]:
github.com/minio/minio/pkg/iam/openid.(*JWKSArgs).PopulatePublicKey(0xc00058eb68, 0x0, 0x0)
        /q/.q/sources/gopath/src/github.com/minio/minio/pkg/iam/openid/jwt.go:75 +0x44c
github.com/minio/minio/pkg/iam/openid.LookupConfig(0xc000dc5580, 0xc000e26ae0, 0x0, 0x0, 0xc00087a640, 0x1a03958, 0x3, 0x0, 0xc0007c8d20, 0x7, ...)
        /q/.q/sources/gopath/src/github.com/minio/minio/pkg/iam/openid/jwt.go:219 +0xcd
github.com/minio/minio/cmd.(*serverConfigV33).lookupConfigs(0xc000a20c00)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/config-current.go:310 +0x5c6
github.com/minio/minio/cmd.newSrvConfig(0x1c95fc0, 0xc000797ea0, 0x1c229a0, 0xc00009c0c0)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/config-current.go:506 +0x34
github.com/minio/minio/cmd.migrateConfigToMinioSys(0x1c95fc0, 0xc000797ea0, 0x1c25840, 0xc000e4c2d0)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/config-migrate.go:2474 +0x74c
github.com/minio/minio/cmd.initConfig(0x1c95fc0, 0xc000797ea0, 0x2, 0x0)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/config.go:181 +0xc4
github.com/minio/minio/cmd.(*ConfigSys).Init(0x2df8bb0, 0x1c95fc0, 0xc000797ea0, 0x0, 0x0)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/config.go:144 +0x211
github.com/minio/minio/cmd.serverMain(0xc00021ef00)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/server-main.go:342 +0x815
github.com/minio/cli.HandleAction(0x1648880, 0x1a03748, 0xc00021ef00, 0xc000233c00, 0x0)
        /q/.q/sources/gopath/pkg/mod/github.com/minio/cli@v1.21.0/app.go:500 +0x7c
github.com/minio/cli.Command.Run(0x1978272, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x199dc78, 0x1b, 0x0, ...)
        /q/.q/sources/gopath/pkg/mod/github.com/minio/cli@v1.21.0/command.go:221 +0x991
github.com/minio/cli.(*App).Run(0xc0002511e0, 0xc0000b6040, 0x4, 0x4, 0x0, 0x0)
        /q/.q/sources/gopath/pkg/mod/github.com/minio/cli@v1.21.0/app.go:261 +0x6ab
github.com/minio/minio/cmd.Main(0xc0000b6040, 0x4, 0x4)
        /q/.q/sources/gopath/src/github.com/minio/minio/cmd/main.go:161 +0x89
main.main()
        /q/.q/sources/gopath/src/github.com/minio/minio/main.go:37 +0x45
```

We've got 2 container per server. Only one of them is running. The other one is crashing. Might be coincidence:

```
# docker service ls
ID                  NAME                MODE                REPLICAS            IMAGE                                       PORTS
jp84629erqwn        minio_haproxy       global              3/3                 127.0.0.1:5000/minio-haproxy:latest   *:80->80/tcp, *:443->443/tcp, *:32000->32000/tcp
jxe1n3x0s0i1        minio_node1         replicated          1/1                 minio/minio:RELEASE.2019-10-11T00-38-09Z
m4vmtk44w0kn        minio_node2         replicated          0/1                 minio/minio:RELEASE.2019-10-11T00-38-09Z
ip17fsgudgv0        minio_node3         replicated          0/1                 minio/minio:RELEASE.2019-10-11T00-38-09Z
y47s3ctx2rac        minio_node4         replicated          1/1                 minio/minio:RELEASE.2019-10-11T00-38-09Z
ocp5s1s5uo32        minio_node5         replicated          0/1                 minio/minio:RELEASE.2019-10-11T00-38-09Z
lvtvyn5n6ugi        minio_node6         replicated          1/1                 minio/minio:RELEASE.2019-10-11T00-38-09Z
```

--- a/pkg/iam/openid/jwt.go
+++ b/pkg/iam/openid/jwt.go
@@ -62,14 +62,11 @@ func (r *JWKSArgs) PopulatePublicKey() error {
 		return err
 	}
 
-	r.publicKeys = make(map[string]crypto.PublicKey)
 	for _, key := range jwk.Keys {
-		var publicKey crypto.PublicKey
-		publicKey, err = key.DecodePublicKey()
+		r.publicKeys[key.Kid], err = key.DecodePublicKey()
 		if err != nil {
 			return err
 		}
-		r.publicKeys[key.Kid] = publicKey
 	}
 
 	return nil
@@ -215,10 +212,18 @@ func LookupConfig(args JWKSArgs, transport *http.Transport, closeRespFn func(io.
 	if err != nil {
 		return args, err
 	}
-	args.URL = u
-	if err := args.PopulatePublicKey(); err != nil {
+
+	args = JWKSArgs{
+		URL:         u,
+		publicKeys:  make(map[string]crypto.PublicKey),
+		transport:   transport,
+		closeRespFn: closeRespFn,
+	}
+
+	if err = args.PopulatePublicKey(); err != nil {
 		return args, err
 	}
+
 	return args, nil
 }
 

